### DISCUSSION TITLE
Efficiencies in my config moving to Gmail Processor

### DISCUSSION THREAD

**Discussion by matt-jon35** [ahochsteger/gmail-processor#325] [Q&A] [answered]
Hello,

I had previously been using Gmail2Drive and am nearly ready to migrate to Gmail Processor, but feel my approach could be better.

When converting my old gmail2drive script to the new format the config has increased from about 3k lines to about 12k, so I am confident I can streamline this somehow.

The workflow :

1. User submits a form
2. A PDF is generated and emailed to a specific address.
3. The attachment has the name format of "School A-Report Type (e.g. Internal)-Reading-Date".  The attachment is then moved into a folder structure in Drive: Home/Reports/School A/Form responses/Report Type/Reading

Can I use variables somehow to streamline this? or a just a generally better approach?

In Gmail2Drive the config I was using always seemed clunky because I had to a new line for each school and each new report type like below:

{ // School A-Internal-Reading
"filter": "-label:to-gdrive-processed -label:archive in:inbox from:noreply@email.com subject:(School A-Internal-Reading-) has:attachment",
"folder": "'Home/Reports/School A/All Reports/Internal/Reading'",
"archive": true
}

Thank you in advance.

**Comment by ahochsteger** [comment]
@matt-jon35 of course there are several ways to streamline the configuration.
The most helpful for you would be to use named regular expression capture groups when matching the attachment name. The documentation for placeholder substitution explains how to do it.
Don't hesitate to ask, if you need more specific information.

**Reply by matt-jon35** [reply]
Hi, thank you so much for reply. Could you provide me with some basic steps for how to achieve this, please, give my example naming convetnion?

Thank you!

**Reply by ahochsteger** [reply, marked as answer]
@matt-jon35 have a look at the [regular example example documentation](https://ahochsteger.github.io/gmail-processor/docs/examples/advanced/regularExpressions) which should cover your use case quite well.
Let me know, if it works or if you need more help.

**Reply by matt-jon35** [reply]
@ahochsteger Thank you so much, I am so appreciative!  I have been able to create exactly what I needed - condensing 12k lines of config down to about 100 and making it so much simpler to work with.

On a side note, how do I know in what order emails being processed?  As i've created to configs: 1 to processed 'new' reports and 1 to process 'edited' reports - if I have received 3 emails with edits will it process the oldest email first?

**Reply by ahochsteger** [reply]
@matt-jon35 wow that's quite impressive, that a reduction of over 99%!
Processing is currently done in this way:
* Process thread/message/attachment configs in the order within the configuration
* Process each list of threads/messages/attachments within each config in the order GMail returns it by default, which seems to be newest first.
Feel free to open a feature request to control the processing order in the config, which seems to be required in your case.

**Reply by matt-jon35** [reply]
Thank you. Your support has been great.
